app_dart/lib/src/service/scheduler.dart

```
@@ -227,11 +247,14 @@ class Scheduler {
       commit.slug,
       '.ci.yaml',
       httpClientProvider: httpClientProvider,
-      ref: commit.sha!,
+      ref: commit.sha == null ? Config.defaultBranch(commit.slug) : commit.sha!,
```

ditto: sha shouldn't be null. If it is we need to throw an exception